mm/slub.c

```diff
@@ -1484,12 +1484,10 @@ static inline bool slab_free_freelist_hook(struct kmem_cache *s,
 	void *old_tail = *tail ? *tail : *head;
 	int rsize;
 
-	/*
-	 * TODO: slab_free_freelist_hook() doesn't work with KFENCE-allocated
-	 * addresses. Need to handle them separately.
-	 */
-	if (is_kfence_address(*head))
+	if (is_kfence_address(next)) {
+		slab_free_hook(s, next);
```

Comment:
I haven't tested with SLAB yet, but I assume it has the same problem.

Reply:
I've been looking into removing this check altogether.

Reply:
But this is also fine.